branch/git.py

```diff
@@ -148,7 +153,7 @@ def from_string(line):
         parents = LogEntry._build_parents(matcher.group(2))
         branches = LogEntry._build_branches(matcher.group(3))
         message = matcher.group(4)
-        is_head = LogEntry.HEAD_MARKER in matcher.group(3)
+        is_head = 'HEAD' in branches or LogEntry.HEAD_MARKER in matcher.group(3)
```

Comment:
This will not work if there is a branch named `HEAD`. Add support for ambiguous HEAD.